### ISSUE TITLE
Math.max should preserve IndexFor

### ISSUE THREAD

**Issue by panacekcz** [panacekcz/checker-framework#20]
When two indices are merged using `Math.max`, the result is also an index.
```java
void m(int[] a, @IndexFor("#1") int i1, @IndexFor("#1") int i2){
 int v = a[Math.max(i1,i2)];
}
```
Will a polymorphic qualifier work?

**Comment by panacekcz** [comment]
A solution based on polymorphic qualifier or special handling in the checker should be easy.
Might also apply to `Math.min` and others.

**Comment by panacekcz** [comment]
`Math.min` already has a special support.
https://github.com/typetools/checker-framework/blob/951cb42b38faac05d9419ade07be87893643dd5c/checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java#L411

**Comment by panacekcz** [comment]
Fixed by typetools#1472.
